File: javascript/bob/bob.js
export const hey = x => {
  let isYelling = x.toUpperCase() == x && x.toLowerCase() != x
  let isNothing = x.trim() == ""
  let isQuestion = [...(x.trim())].pop() == '?'

  if (isNothing) return 'Fine. Be that way!'
  if (isYelling && isQuestion) return "Calm down, I know what I'm doing!"
  if (isYelling) return "Whoa, chill out!"
  if (isQuestion) return "Sure."
  return "Whatever."
}

File: haskell/bob/src/Bob.hs
module Bob (responseFor) where

import Data.String.Utils
import Data.Char

responseFor :: String -> String
responseFor x
  | isNothing = "Fine. Be that way!"
  | isYelling && isQuestion = "Calm down, I know what I'm doing!"
  | isYelling = "Whoa, chill out!"
  | isQuestion = "Sure."
  | otherwise = "Whatever."
  where isYelling = map toUpper x == x && map toLower x /= x
        isQuestion = last (strip x) == '?'
        isNothing = strip x == ""
